File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/Grid.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

@Builder
public record Grid(
    String id,
    List<GridContent> content,
    Page<?> page,
    boolean tree,
    int size,
    String style,
    String cssClasses,
    boolean wrapCellContent,
    boolean compact,
    boolean noBorder,
    boolean noRowBorder,
    boolean columnBorders,
    boolean rowStripes,
    String vaadinGridCellBackground,
    String vaadinGridCellPadding)
    implements Component {}

File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/componentmetadata/Grid.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import { Page } from "@mateu/shared/apiClients/dtos/Page";
import ClientSideComponent from "@mateu/shared/apiClients/dtos/ClientSideComponent";

export default interface Grid extends ComponentMetadata {

    content: ClientSideComponent[]
    page: Page
    tree: boolean

}